src/Cache/BaseDataCache.php

```diff
@@ -50,15 +50,16 @@ public function get_data(string $key, $default = null)
             return $default;
         }
 
-        // ignore data if internal cache expiration time is not set
-        if (!array_key_exists('__cache_expiration_time', $data)) {
-            return $default;
-        }
-
-        // ignore data if internal cache expiration time is expired
-        if ($data['__cache_expiration_time'] < time()) {
-            return $default;
-        }
+        // FreshRSS commented out, to allow HTTP 304
+        // // ignore data if internal cache expiration time is not set
+        // if (!array_key_exists('__cache_expiration_time', $data)) {
+        //     return $default;
+        // }
+
+        // // ignore data if internal cache expiration time is expired
+        // if ($data['__cache_expiration_time'] < time()) {
+        //     return $default;
+        // }
```

Comment:
@Art4 This looks suspicious, probably broken in the current version of SimplePie. This seems to disable the possibility of conditional requests (HTTP 304).